amp-viewer.js

```diff
@@ -8,116 +8,123 @@
  * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
  */
 
-(function(win) {
-  'use strict';
+'use strict'; {
 
-  document.registerElement('amp-viewer', {
-    prototype: {
-      __proto__: HTMLElement.prototype,
+  customElements.define('amp-viewer', class extends HTMLElement {
 
-      createdCallback: function() {
-        this._amp = null;
-        this._src = '';
-        this._host = null;
+    constructor() {
+      super();
 
-        if (!win.AMP_SHADOW) {
-          win.AMP_SHADOW = true;
-          this._installScript('https://cdn.ampproject.org/shadow-v0.js');
-        }
+      this._amp = null;
+      this._src = '';
+      this._host = null;
+    }
 
-        if (this.hasAttribute('src')) {
-          this.src = this.getAttribute('src');
-        }
-      },
+    connectedCallback() {
 
-      set src(src) {
-        if (this._src === src) {
-          return;
-        }
+      if (!window.AMP_SHADOW) {
```

Comment:
This was previously done in `createdCallback`, which corresponds to `constructor` in Custom Elements v1. This `if` should be moved up in the constructor not to change the timing